### ISSUE TITLE
id_list1 undefined

### ISSUE THREAD

**Issue by mdelerue** [PokemonGoF/PokemonGo-Bot#267]
### Actual Behavior

Seems to happen when pokemons are exchanged for candy :
```
    Traceback (most recent call last):
    File "pokecli.py", line 107, in <module>main()
    File "pokecli.py", line 104, in main bot.take_step()
    File "/root/PokemonGo-Bot/pokemongo_bot/__init__.py", line 31, in take_step self.stepper.take_step()
   File "/root/PokemonGo-Bot/pokemongo_bot/stepper.py", line 58, in take_step
    self._work_at_position(position[0], position[1], position[2], True)
  File "/root/PokemonGo-Bot/pokemongo_bot/stepper.py", line 100, in _work_at_position
    self.bot.work_on_cell(cell, position, pokemon_only)
  File "/root/PokemonGo-Bot/pokemongo_bot/__init__.py", line 38, in work_on_cell
    worker.work()
  File "/root/PokemonGo-Bot/pokemongo_bot/cell_workers/pokemon_catch_worker.py", line 86, in work
   self.transfer_pokemon(list(Set(id_list2) - Set(id_list1)))
   NameError: global name 'id_list1' is not defined
```

### Steps to Reproduce
Just wait for the first exchange

### Other Information
OS: Debian Jessie.
Dev Branch
acc2d4f6624b48468d58078520adbb53091a8b64

**Comment by rkodali22** [comment]
Look at PR  #254

**Comment by mdelerue** [comment]
OK, thanks. 
I pulled, it is better (even if they are still in inventory)

**Comment by douglascamata** [comment]
Fixed in dev now by abd2d713a917e8325b5b438ffc44b1a60ecceac8

**Comment by mdelerue** [comment]
Thank you guys
